python/ray/autoscaler/autoscaler.py

```diff
@@ -428,7 +428,7 @@ def _get_node_type_specific_fields(self, node_id: str,
     def _get_node_specific_docker_config(self, node_id):
         docker_config = copy.deepcopy(self.config.get("docker", {}))
         node_specific_docker = self._get_node_type_specific_fields(
-            node_id, "docker")
+            node_id, "docker", {})
```

Comment:
Can we just do a none-check here instead of the other changes?

Reply:
for reference, all the other fields are guaranteed to exist in the different node provider configs, and are covered under unit tests

Reply:
SGTM